source/argument_parser.py

```diff
@@ -0,0 +1,86 @@
+import argparse
+
+from interface import (ArgumentParserInterface, InputReaderInterface, OutputFormatterInterface,
+                       TaskExecutorInterface, Arguments, LibraryComponents)
+
+from source import (DirectoryReader, DirectoryWatcher, OutputFormatterFasta,
+                    OutputFormatterFastq, SequentialTaskExecutor, ParallelTaskExecutor)
+
+
+class ArgumentParser(ArgumentParserInterface):
+
+    def __init__(self) -> None:
+
+        self.parser = argparse.ArgumentParser(description='Fast caller for ONT reads')
+        self.library_components = None
+
+    def parse_arguments(self) -> Arguments:
+
+        self.parser.add_argument('--directory', type=str, nargs='*',
```

Comment:
this is weird that you add arguments to parser only in parse_args

Reply:
I'm not sure I understand what you mean.

Reply:
right now, parse_arguments is not *idempotent* despite it's name suggesting that it does not mutate anything. Try to run it twice and see the fireworks happen ...

Reply:
Functional design should fix this one.